tests/unit/src/main/scala/tests/JVMVersion.scala

```diff
@@ -0,0 +1,117 @@
+package tests
+
+import scala.util._
+import scala.util.matching._
+
+/** ADT used to model the current version of the runtime JVM.
+ *
+ * It is needed to know the current version for several of the test cases as
+ * there are some changes to the classpath between different JVM versions.
+ *
+ * For example, on JDK > 8 several standard jar files are now stored as java
+ * modules and will now longer show up as standalone jars,
+ * e.g. `charsets.jar`.
+ */
+sealed trait JVMVersion extends Product with Serializable {
```

Comment:
Did you try using `scala.util.Properties.isJavaAtLeast`?
```scala
(Scala 2.12.9 Java 1.8.0_222)
@ scala.util.Properties.isJavaAtLeast("9")
res1: Boolean = false


(Scala 2.12.9 Java 11.0.5)
@ scala.util.Properties.isJavaAtLeast("11")
res2: Boolean = true
```

Reply:
I did not, I did not know about that. I've updated the code, removed the `JVMVersion` class and am now using that predicate.